src/order/basic.lean

```diff
@@ -369,6 +442,77 @@ lemma monotone [partial_order α] [preorder β] {f : α → β} (H : strict_mono
 
 end strict_mono
 
+namespace strict_anti
+open ordering function
+
+protected lemma strict_anti_on [has_lt α] [has_lt β] {f : α → β} (hf : strict_anti f)
+  (s : set α) :
+  strict_anti_on f s :=
+λ x hx y hy hxy, hf hxy
+
+
+lemma comp [has_lt α] [has_lt β] [has_lt γ] {g : β → γ} {f : α → β}
+  (hg : strict_anti g) (hf : strict_anti f) :
+  strict_mono (g ∘ f) :=
+λ a b h, hg (hf h)
+
+lemma comp_strict_mono [has_lt α] [has_lt β] [has_lt γ] {g : β → γ} {f : α → β}
+  (hg : strict_anti g) (hf : strict_mono f) :
+  strict_anti (g ∘ f) :=
+λ a b h, hg (hf h)
+
+protected lemma ite' [preorder α] [has_lt β] {f g : α → β} (hf : strict_anti f) (hg : strict_anti g)
+  {p : α → Prop} [decidable_pred p] (hp : ∀ ⦃x y⦄, x < y → p y → p x)
+  (hfg : ∀ ⦃x y⦄, p x → ¬p y → x < y → g y < f x) :
+  strict_anti (λ x, if p x then f x else g x) :=
```

Comment:
Can you use the `order_dual` trick here to avoid repeating the proof?

Reply:
I can, but that involves changing the hypotheses so that `p` is upward-closed rather than downward. If you think that's fine, then sure.

Reply:
I don't understand your comment here. Doesn't
```
@strict_mono.ite' α (order_dual β) _ _ f g hf hg p _ hp hfg
```
work fine, without any change to the assumptions?

Reply:
Hmm... yes it does. I was thinking about dualizing `α`, not `β`.